### PR TITLE
Extract EEI methods from within callImport (call, create, selfDestruct, returnDataCopy)

### PR DESCRIPTION
**Pull request by axic**
Part of #156.

Inspired by #148.

### CODE REVIEW

File: src/eei.cpp [hunk 1]
```diff
@@ -722,8 +755,18 @@ inline int64_t maxCallGas(int64_t gas) {
...
+      case EEICallKind::Call: methodName = "call"; break;
+      case EEICallKind::CallCode: methodName = "callCode"; break;
+      case EEICallKind::CallDelegate: methodName = "callDelegate"; break;
+      case EEICallKind::CallStatic: methodName = "callStatic"; break;
+      }
+#endif
+
       HERA_DEBUG <<
-        import->base << " " << hex <<
+        methodName << " " << hex <<
```
Comment (jakelang) [inline src/eei.cpp]: this needs to be conditionally compiled with HERA_DEBUG for release builds to work correctly.

Comment (axic) [reply src/eei.cpp]: That's fine, the main question in this PR is the naming and general idea.

File: src/eei.cpp [hunk 2]
```diff
@@ -659,6 +657,40 @@ namespace hera {
...
+    heraAssert(false, string("Unsupported import called: ") + import->module.str + "::" + import->base.str + " (" + to_string(arguments.size()) + "arguments)");
+  }
+
+  void EthereumInterface::eeiReturnDataCopy(uint32_t dataOffset, uint32_t offset, uint32_t size)
+  {
+      safeChargeDataCopy(size, GasSchedule::verylow);
+      storeMemory(m_lastReturnData, offset, dataOffset, size);
+  }
+
+  uint32_t EthereumInterface::eeiCall(EEICallKind kind, int64_t gas, uint32_t addressOffset, uint32_t valueOffset, uint32_t dataOffset, uint32_t dataLength)
```
Comment (jakelang) [inline src/eei.cpp]: I think it may make sense to trade some code repetition for more clarity here, as in #148 where they are split.

Comment (axic) [reply src/eei.cpp]: That was the main point of enum so that it can be shared 😉

File: src/eei.h [hunk 3]
```diff
@@ -76,6 +76,8 @@ struct EthereumInterface : ShellExternalInterface {
 
   // EEI methods
 
+  uint32_t eeiGetReturnDataSize();
```
Comment (jakelang) [inline src/eei.h]: Refer to #148 for the ordering of these function declarations. I think from a maintainability perspective it makes sense to have a more logical ordering of the methods (the design doc needs to also be re-ordered).

Comment (axic) [reply src/eei.h]: Yeah, would need to apply that to both design, hera, ewasm-rust-api, as-ewasm-api.

But perhaps not on this PR.

**Pull request merged**